rllib/agents/a3c/a3c_tf_policy.py

```diff
@@ -1,178 +1,181 @@
 """Note: Keep in sync with changes to VTraceTFPolicy."""
-from typing import Optional, Dict
 import gym
+from typing import Dict, List, Type, Union
 
 import ray
 from ray.rllib.policy.sample_batch import SampleBatch
-from ray.rllib.evaluation.episode import Episode
 from ray.rllib.evaluation.postprocessing import (
     compute_gae_for_sample_batch,
     Postprocessing,
 )
 from ray.rllib.models.action_dist import ActionDistribution
 from ray.rllib.models.modelv2 import ModelV2
+from ray.rllib.models.tf.tf_action_dist import TFActionDistribution
+from ray.rllib.policy.dynamic_tf_policy_v2 import DynamicTFPolicyV2
+from ray.rllib.policy.eager_tf_policy_v2 import EagerTFPolicyV2
 from ray.rllib.policy.policy import Policy
 from ray.rllib.policy.tf_mixins import (
+    compute_gradients,
     EntropyCoeffSchedule,
     LearningRateSchedule,
     ValueNetworkMixin,
 )
-from ray.rllib.policy.tf_policy_template import build_tf_policy
+from ray.rllib.utils.annotations import override
 from ray.rllib.utils.deprecation import Deprecated
 from ray.rllib.utils.framework import try_import_tf
 from ray.rllib.utils.tf_utils import explained_variance
 from ray.rllib.utils.typing import (
-    TrainerConfigDict,
-    TensorType,
-    PolicyID,
     LocalOptimizer,
     ModelGradients,
+    TensorType,
 )
 
 tf1, tf, tfv = try_import_tf()
 
 
+# We need this builder function because we want to share the same
+# custom logics between TF1 dynamic and TF2 eager policies.
+def get_a3c_tf_policy(base: Type[Union[DynamicTFPolicyV2, EagerTFPolicyV2]]) -> Type[Union[DynamicTFPolicyV2, EagerTFPolicyV2]]:
```

Comment:
ah, this type annotation is very nice.
do you think we should create a shortcut name in utils.typing, so it reads a bit shorter here?

Reply:
will do!

Reply:
done